Connect MDX components, add additional stories & datasets, update homepage, and add assets

sandrahoang686 opened the pull request:
## Available PR templates
**Related Issues:** 
* https://github.com/NASA-IMPACT/veda-ui/issues/1324
* https://github.com/NASA-IMPACT/veda-ui/issues/1313

**Related PR:** https://github.com/NASA-IMPACT/veda-ui/pull/1358

This work is an extension off of @aboydnw [original PR](https://github.com/developmentseed/next-veda-ui/pull/22) that adds content.

**Changes:**
* Use and connect more exposed MDX Components from veda-ui
* Adding 2 stories, 3 datasets, and supporting files 

**Related PRs:**
https://github.com/developmentseed/next-veda-ui/pull/34

sandrahoang686 commented:
@aboydnw closed the [original PR](https://github.com/developmentseed/next-veda-ui/pull/22) in favor of this one and left a comment on the closed PR on why. But re-iterating here that I did this because to debug the various bugs, I had to add content 1 by 1 and fix each issue as I went so this was easiest for me. If you can confirm that i've added all the content you had originally in the old PR, thanks.

hanbyul-here commented on app/components/mdx-components/block.tsx:
```diff
@@ -14,4 +14,20 @@ export default function EnhancedMapBlock(props) {
+export function EnhancedScrollyTellingBlock(props) {
+  const { datasets } = useDataStore();
+  const transformed = {};
+  datasets?.map((dataset) => {
```
👋 Considering Mapblock and Scrollytelling block expects the same type of the data, (My PR here might be helpful: https://github.com/NASA-IMPACT/veda-ui/pull/1361) but we have to pass different shape of the data to each block, I think something is wrong. Would you be able to investigate what is going on? (Or am I missing anything?)

sandrahoang686 replied:
I reviewed your PR and left a comment but made some changes to get it merged in. We are missing something because the same data structure breaks. I'll have to investigate further.

sandrahoang686 replied:
Okay this has been figured out, mapblock works when passing in `DatasetData[]` or `VedaData<DatasetData>` so both types but it actually expects `VedaData<DatasetData>` just like scrollytelling, so i've consolidated. I've also exposed these types from veda-ui and use them here. This is ready for re-review 🙏🏼 .

aboydnw reviewed (commented):
I added a few tweaks to [this ticket](https://github.com/NASA-IMPACT/veda-ui/issues/1282) so we don't lose track of them, but I think we can address them later

Only thing I am seeing with this PR is that only the first frame of the scrollytelling block is loading for me in the[ air quality story](https://deploy-preview-34--veda-ui-next-test.netlify.app/stories/air-quality-and-covid-19). The content and frames seem to all be there, but the only frame that loads anything in the map (basemap or layer) is the first one.

sandrahoang686 commented:
> I added a few tweaks to [this ticket](https://github.com/NASA-IMPACT/veda-ui/issues/1282) so we don't lose track of them, but I think we can address them later
> 
> Only thing I am seeing with this PR is that only the first frame of the scrollytelling block is loading for me in the[ air quality story](https://deploy-preview-34--veda-ui-next-test.netlify.app/stories/air-quality-and-covid-19). The content and frames seem to all be there, but the only frame that loads anything in the map (basemap or layer) is the first one.

@aboydnw i addressed that [here](https://github.com/developmentseed/next-veda-ui/pull/34#pullrequestreview-2538341715) but if thats a blocker and want it shipped with this PR, lmk 👍🏼

aboydnw commented:
Connected with Sandra in slack and she let me know the issue I spotted is what was meant by scrollytelling styling in her original comment, and will be addressed with https://github.com/NASA-IMPACT/veda-ui/issues/1282

I adjusted the language there to be more specific and we can address that later

hanbyul-here commented on public/images/homepage/user-enthusiasts.jpg:
Did you mean to just add assets?

sandrahoang686 replied:
yeah, i told anthony that i'll keep the assets here since homepage will be added anyways
